ext/intl/dateformat/dateformat_create.cpp

```diff
@@ -112,7 +112,7 @@ static zend_result datefmt_ctor(INTERNAL_FUNCTION_PARAMETERS, zend_error_handlin
 	}
 	locale = Locale::createFromName(locale_str);
 	/* get*Name accessors being set does not preclude being bogus */
-	if (locale.isBogus() || strlen(locale.getISO3Language()) == 0) {
+	if (locale.isBogus() || (strcmp("C", locale_str) && strlen(locale.getISO3Language()) == 0)) {
```

Comment:
```suggestion
	if (locale.isBogus() || (locale_str == 1 && locale_str[0] == 'C') && strlen(locale.getISO3Language()) == 0)) {
```
The call to strcmp() is not needed.

Reply:
I guess you meant locale_len == 1

Reply:
Yes indeed